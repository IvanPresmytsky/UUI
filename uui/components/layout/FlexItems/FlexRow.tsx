import { FlexRowProps as uuiFlexRowProps, withMods } from '@epam/uui-core';
import { FlexRow as uuiFlexRow } from '@epam/uui-components';
import css from './FlexRow.module.scss';

export type RowMods = {
    /** Defines row size */
    size?: null | '24' | '30' | '36' | '42' | '48';
    /** Pass true, to enable row bottom border */
    borderBottom?: boolean;
    /** Defines row column gap */
    columnGap?: number | '6' | '12' | '18' | '24' | '36';
    /** Defines row margin */
    margin?: '12' | '24';
    /** Defines horizontal row padding */
    padding?: '6' | '12' | '18' | '24';
    /** Defines row gap */
    rowGap?: number | '6' | '12' | '18' | '24' | '36';
    /** Defines row spacing */
    spacing?: '6' | '12' | '18';
    /** Pass true, to show a top shadow */
    topShadow?: boolean;
    /** Defines vertical row padding */
    vPadding?: '12' | '18' | '24' | '36' | '48';
    /** Defines row background */
    background?: 'surface-main';
};

/** Represents the properties of the FlexRow component. */
export interface FlexRowProps extends Omit<uuiFlexRowProps, 'columnGap' | 'rowGap'>, RowMods {}

export const FlexRow = withMods<Omit<uuiFlexRowProps, 'columnGap' | 'rowGap'>, RowMods>(uuiFlexRow, (props) => {
    return [
        css.root,
        props.size !== null && css['size-' + (props.size || '36')],
        props.padding && css['padding-' + props.padding],
        props.vPadding && css['vPadding-' + props.vPadding],
        props.margin && css['margin-' + props.margin],
        props.topShadow && css.topShadow,
        props.borderBottom && css.borderBottom,
        props.spacing && css['spacing-' + props.spacing],
        props.background && css[`uui-${props.background}`],
    ];
});
