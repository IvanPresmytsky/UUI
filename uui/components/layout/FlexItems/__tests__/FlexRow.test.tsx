import React from 'react';
import { FlexRow } from '../FlexRow';
import { renderer } from '@epam/uui-test-utils';

describe('FlexRow', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<FlexRow />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const tree = renderer
            .create(<FlexRow 
                onClick={ () => {} } 
                margin="12" 
                size="24" 
                padding="12" 
                borderBottom 
                alignItems="top" 
                topShadow 
                vPadding="12" 
                rowGap="24"
                columnGap="24"
                borderTop={ true }
                justifyContent="center"
                cx="some class"
                rawProps={ { id: '123' } }
                key="key"
                background="surface-main"
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
