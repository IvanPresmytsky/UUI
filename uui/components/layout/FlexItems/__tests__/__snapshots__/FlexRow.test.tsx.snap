// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexRow should be rendered correctly 1`] = `
<div
  className="root uui-flex-row uui-size-36"
  style={Object {}}
/>
`;

exports[`FlexRow should be rendered correctly with props 1`] = `
<div
  className="root uui-flex-row -clickable some class uui-size-24 alignItems justifyContent padding vPadding margin topShadow borderBottom borderTop columnGap rowGap uui-flex-row-bg-surface-main"
  id="123"
  onClick={[Function]}
  style={
    Object {
      "--uui-flex-row-align-items": "top",
      "--uui-flex-row-column-gap": "24px",
      "--uui-flex-row-justify-content": "center",
      "--uui-flex-row-margin": "12px",
      "--uui-flex-row-padding": "12px",
      "--uui-flex-row-row-gap": "24px",
      "--uui-flex-row-v-padding": "12px",
    }
  }
/>
`;
